### ISSUE TITLE
[feature request] For hierarchical reduction, add an option to combine the most similar topics regardless of size

### ISSUE THREAD

**Issue by MackieBlackburn** [ddangelov/Top2Vec#83]
It seems like the approach of combining the smallest topics with the ones they are most similar to has some benefits but could potentially ignore lower-volume topics that are really distinct. Maybe there should be an option to iteratively combine the topics that are most similar to each other rather than starting with the smallest topics.

**Comment by ddangelov** [comment]
This has already been experimented with quite extensively and results have been unsatisfactory. The motivation behind the current approach is to find the most representative topics of a corpus. Smaller topics, by virtue of their size are less important. 

That being said, I plan to work extensively on improving hierarchical reduction so perhaps some reasonable tradeoff can be found between merging based on similarity and size. Improvements will be coming.

**Comment by MackieBlackburn** [comment]
It depends on the application but you can't always make the assumption that smaller topics are less important.

**Comment by ddangelov** [comment]
Agreed, it depends on the motivations for the topic reduction. Factoring in the distinctiveness of topics during the reduction will certainly be factored in for the future version. Thank you for your input.

**Comment by MackieBlackburn** [comment]
I probably don't need to create another issue for this but I've found that the top words for each topic often contain redundant information or rare misspellings (e.g. "government", "goverment", "governent") I think these kind of words may be very close to the topic vector but they are also very rare or redundant. I found that weighting the words by TF-IDF or frequency within documents tagged to the topic may be more descriptive and representative, or at least they look better when reviewing the topics.
